### PR TITLE
New vusb dameon

### PR DESCRIPTION
**Pull request by jean-edouard**
Use this pull request to comment on the code.

See the main pull request for more information:
https://github.com/OpenXT/xenclient-oe/pull/146

### CODE REVIEW

File: src/db.c [hunk 1]
```diff
@@ -0,0 +1,390 @@
...
+
+xcdbus_conn_t *db_xcbus = NULL; /**< A dbus (libxcdbus) handle, initialized by db_init() */
+
+static char*
+parse_value(char *node_path, char *key)
+{
+  char *value;
+  char path[128];
+
+  sprintf(path, "%s/%s", node_path, key);
```
Comment (cjp256) [inline src/db.c]: snprintf would be strongly preferred here (this comment applies to remaining uses of sprintf as well)

File: src/policy.c [hunk 2]
```diff
@@ -0,0 +1,421 @@
...
+            continue;
+          } else {
+            /* The device is already assigned to the right VM */
+            continue;
+          }
+        } else {
+          /* The device is not assigned, as expected, plug it to its VM */
+          /* No need to check the policy, ALWAYS implies ALLOW */
+          device->vm = vm;
+          ret |= usbowls_plug_device(vm->domid, device->busid, device->devid);
```
Comment (cjp256) [inline src/policy.c]: docs say usbowls_plug_device returns 1 on failure, yet this function is supposed to return -1. I'm not sure what's happening here, but I suspect something is amiss :)

File: src/udev.c [hunk 3]
```diff
@@ -0,0 +1,590 @@
...
+  /* Sysname must start with a digit */
+  udev_enumerate_add_match_sysname(enumerate, "[0-9]*");
+  udev_enumerate_scan_devices(enumerate);
+  udev_device_list = udev_enumerate_get_list_entry(enumerate);
+  udev_list_entry_foreach(udev_device_entry, udev_device_list) {
+    path = udev_list_entry_get_name(udev_device_entry);
+    udev_device = udev_device_new_from_syspath(udev_handle, path);
+    if (udev_maybe_add_device(udev_device, 0) == NULL)
+      udev_device_unref(udev_device);
+    else
```
Comment (cjp256) [inline src/udev.c]: else-then-notta? perhaps put the "else" into the comment to get the point across without looking like an error?

**Comment by jean-edouard** [comment]
Previous code comments should be addressed in the latest commit.

**Comment by rossphilipson** [comment]
Sounds like folks are in agreement that it is time to get this work in. I will merge this work in a while in case there are any last minute comments.

**Pull request merged**